networkx/convert.py

```diff
@@ -98,7 +98,7 @@ def to_networkx_graph(data, create_using=None, multigraph_input=False):
             raise nx.NetworkXError("Input is not a correct pygraphviz graph.") from err
 
     # dict of dicts/lists
-    if isinstance(data, dict):
+    if hasattr(data, "setdefault"):
```

Comment:
For builtin modules I think it's better to use `isinstance` since that better identifies the type of an object. It is possible that something that isn't a dict has `setdefault`.
For example, in the check for `format` (for scipy) it can crash when passing a `str`. It's not supported but it's crashing at the wrong place.

For external libraries (pandas/numpy) what about checking the `__module__` or `__class__` name of the object instead?

For example:
```
if data.__module__.startswith("pandas."):
    import pandas
    do stuff with pandas
```